### PR TITLE
Update UNK firmware configuration.

### PR DESCRIPTION
**Pull request by herpiko**
Update the keyboard configuration to sync with the latest QMK API.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/unk/info.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
+{
+    "keyboard_name": "UNK",
+    "url": "https://github.com/herpiko/unk",
+    "maintainer": "herpiko",
+    "width": 17,
+    "height": 5,
```
Comment (waffle87) [inline keyboards/handwired/unk/info.json:6]:
```suggestion
```

File: keyboards/handwired/unk/keymaps/default/config.h [hunk 2]
```diff
@@ -18,6 +18,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #pragma once
 
+#define USE_SERIAL
+
```
Comment (waffle87) [inline keyboards/handwired/unk/keymaps/default/config.h:22]: This isn't needed.
```suggestion
```

File: keyboards/handwired/unk/rev1/config.h [hunk 3]
```diff
@@ -30,6 +44,26 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* number of backlight levels */
+// #define BACKLIGHT_LEVELS 3
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
```
Comment (waffle87) [inline keyboards/handwired/unk/rev1/config.h:66]: Why are you adding this back? This is all present in `info.json`, and does not belong in `config.h` anymore.

File: keyboards/handwired/unk/rules.mk [hunk 4]
```diff
@@ -1,16 +1,40 @@
...
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
```
Comment (waffle87) [inline keyboards/handwired/unk/rules.mk:13]: Same goes for this file...
```suggestion
```

File: keyboards/handwired/unk/rules.mk [hunk 5]
```diff
@@ -1,16 +1,40 @@
...
 # Build Options
 #   change yes to no to disable
 #
-BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (waffle87) [inline keyboards/handwired/unk/rules.mk:17]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
```

File: keyboards/handwired/unk/rules.mk [hunk 6]
```diff
@@ -1,16 +1,40 @@
...
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
 RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
 AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
 
 SPLIT_KEYBOARD = yes
+SPLIT_COMMUNICATION = serial
+
```
Comment (waffle87) [inline keyboards/handwired/unk/rules.mk:35]:
```suggestion
SPLIT_KEYBOARD = yes
```

File: keyboards/handwired/unk/rules.mk [hunk 7]
```diff
@@ -1,16 +1,40 @@
...
 
 DEFAULT_FOLDER = handwired/unk/rev1
+OLED_DRIVER_ENABLE = yes
+OLED_ENABLE = yes
+OLED_DRIVER = SSD1306
```
Comment (waffle87) [inline keyboards/handwired/unk/rules.mk:40]:
```suggestion
OLED_ENABLE = yes
```
And if OLEDs are enabled at the keyboard level, there also needs to be some sort of functionality for them at the keyboard level, eg. in a `rev1.c` file.

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by herpiko** [comment]
@zvecr @waffle87 Thank you for pointing it out. I've not touched this for years and just tried to make it compiled. I'll adjust the PR based on your suggestions.

File: keyboards/handwired/unk/keymaps/default/keymap.c [hunk 8]
```diff
@@ -13,41 +13,54 @@ enum custom_keycodes {
   COLEMAK,
 };
 
+#define MK_KINETIC_SPEED
+#define MOUSEKEY_WHEEL_INTERVAL 200
+#define MOUSEKEY_WHEEL_DELAY 0
+#define MOUSEKEY_WHEEL_MAX_SPEED 20
+#define MOUSEKEY_WHEEL_TIME_TO_MAX 1
```
Comment (drashna) [inline keyboards/handwired/unk/keymaps/default/keymap.c:20]: These aren't valid here and won't be picked up.

**Comment by herpiko** [comment]
Hi @zvecr @waffle87 @drashna @technomancy , I have updated the patch. Please kindly review my PR again.

File: keyboards/handwired/unk/rev1/config.h [hunk 9]
```diff
@@ -17,8 +17,16 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+/* USB Device descriptor parameter */
 
-#define MASTER_LEFT // Comment this line for the right half firmware
+/* key matrix size */
+// Rows are doubled-up
+#define MATRIX_ROWS 10
+#define MATRIX_COLS 8
+
+
+
```
Comment (waffle87) [inline keyboards/handwired/unk/rev1/config.h:28]: These were intentionally removed previously.
```suggestion
```

File: keyboards/handwired/unk/rev1/config.h [hunk 10]
```diff
@@ -17,8 +17,16 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+//#define MASTER_LEFT // Comment this line for the right half firmware
```
Comment (waffle87) [inline keyboards/handwired/unk/rev1/config.h:29]:
```suggestion
```

File: keyboards/handwired/unk/rev1/config.h [hunk 11]
```diff
@@ -30,6 +38,22 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+// Because we are using OLED on I2C.
+// USE_I2C does not to be defined here since OLED_DRIVER_ENABLE has been set in rules.mk
+#define SELECT_SOFT_SERIAL_SPEED 1
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* number of backlight levels */
+// #define BACKLIGHT_LEVELS 3
+
```
Comment (waffle87) [inline keyboards/handwired/unk/rev1/config.h:56]: This is all default behaviour.
```suggestion
```

File: keyboards/handwired/unk/rev1/rules.mk [hunk 12]
```diff
@@ -0,0 +1 @@
+OLED_ENABLE = yes
```
Comment (waffle87) [inline keyboards/handwired/unk/rev1/rules.mk:1]: If OLEDs are enabled at the keyboard level, there needs to be some OLED code at the keyboard level using the `oled_init_kb()` & `oled_task_kb()` functions.

**Review by waffle87 — DISMISSED** [review]

**Comment by technomancy** [comment]
@herpiko Please stop pinging me; I have no idea what unk is and cannot comment on your changes.

**Pull request closed without merging**